src/app/(admin)/new_dashboard/guides/page.tsx

```diff
@@ -1,26 +1,45 @@
 "use client";
 import DashboardCard from "../shared/Card";
 import PageContainer from "@/components/PageContainer";
-import { Box } from "@mui/material";
+import { Box, Typography } from "@mui/material";
 import GuideList from "./components/GuideList";
 import SearchBar from "./components/SearchBar";
-import { useSearchGuides, useGuideItems } from "./hooks";
+import { useSearchGuides, useGuides } from "./hooks";
+import { Guide } from "@/api/types/Guide";
+
 const Page = () => {
-  const { guideItems } = useGuideItems();
-  const { searchQuery, filteredGuides, handleSearchChange } =
-    useSearchGuides(guideItems);
+  const { data: guideItems, isLoading, isError, isSuccess } = useGuides();
+
+  let guides: Guide[] = [];
+
+  if (isSuccess && "data" in guideItems) {
+    guides = guideItems.data;
+  }
+
+  const { searchQuery, filteredGuides, handleSearchChange } = useSearchGuides(
+    guides
+  );
 
   return (
     <PageContainer title="Guides">
       <DashboardCard title="Guides">
         <Box>
-          <Box mb={2}>
-            <SearchBar
-              searchQuery={searchQuery}
-              onSearchChange={handleSearchChange}
-            />
-          </Box>
-          <GuideList guides={filteredGuides} />
+          {isLoading && <Typography>Loading...</Typography>}
+          {isError && <Typography>Error loading guides</Typography>}
+          {isSuccess && guides.length > 0 && (
+            <>
+              <Box mb={2}>
+                <SearchBar
+                  searchQuery={searchQuery}
+                  onSearchChange={handleSearchChange}
+                />
+              </Box>
+              <GuideList guideItems={filteredGuides} />
+            </>
+          )}
+          {isSuccess && guides.length === 0 && (
+            <Typography>No guides available</Typography>
+          )}
```

Comment:
Try adding loading skeletons from MUI. Or alternatively use a ciruclar loader placed at the center of the page.
Also why not to display the search bar? You can just disable it while there's no data or it won't even search anything while there's no data.

Add an alert component from MUI to display the error.